conanfile.py

```diff
@@ -160,13 +165,38 @@ def build(self):
         cmake.configure()
         cmake.build()
 
+        if self.options.get_safe("enable_sentry", False):
+            # Upload debug symbols to sentry
+            sentry_project = self.conf.get("user.curaengine:sentry_project", "", check_type=str)
+            sentry_org = self.conf.get("user.curaengine:sentry_org", "", check_type=str)
+            if sentry_project == "" or sentry_org == "":
+                raise ConanInvalidConfiguration("sentry_project or sentry_org is not set")
+            if which("sentry-cli") is None:
+                self.output.warn("sentry-cli is not installed, skipping uploading debug symbols")
+                self.output.warn("sentry-cli is not installed, skipping release creation")
+                return
+
+            if self.settings.os == "Linux":
+                self.output.info("Stripping debug symbols from binary")
+                self.run("objcopy --only-keep-debug --compress-debug-sections=zlib CuraEngine CuraEngine.debug")
+                self.run("objcopy --strip-debug --strip-unneeded CuraEngine")
+                self.run("objcopy --add-gnu-debuglink=CuraEngine.debug CuraEngine")
+
+            self.output.info("Uploading debug symbols to sentry")
+            build_source_dir = self.build_path.parent.parent.as_posix()
+            self.run(f"sentry-cli debug-files upload --include-sources -o {sentry_org} -p {sentry_project} {build_source_dir}")
```

Comment:
I guess you checkd this, but aren't we going to overload Sentry with our debug data ? Are they removed after some time ?

Reply:
artifacts are only uploaded when changed and stored for 90 days after the last use. The amount of data being stored is not that much and should be within allowed usages, but if we run into actual problems we should setup a symbol server. But that will be a tomorrow problem, if it will be a problem at all.